### PR TITLE
[2409] Build a manual entry route for teach first

### PR DESCRIPTION
**Pull request by JonF1**
### Context
https://trello.com/c/Bb42DaNS/2409-s-build-a-manual-entry-route-for-teach-first

### Changes proposed in this pull request
* ~Boolean model method which sets arbitrary provider as a teach first provider~
Add a new persona/provider with a predefined HPITT code
* New route is the same as assessment only
* Teach first radio button on `trainees/new` renders if the current user is from that teach first provider, all other routes are hidden. If not a teach first provider, the button is hidden
* Model test

### Guidance to review
Login using Persona "Teach First", and BAU.

### CODE REVIEW

**Comment by JonF1** [comment]
[[S] Build a manual entry route for teach first](https://trello.com/c/Bb42DaNS/2409-s-build-a-manual-entry-route-for-teach-first)

**Review by jhenderson — APPROVED** [review]

**Comment by darokel** [comment]
We need to hide all the other route options if it's a TF provider so we probs need a conditional wrap for all the options available?

**Comment by defong** [comment]
Will this break persona?

**Comment by archferns** [comment]
> Will this break persona?

Added a separate HPITT Provider/Persona

File: app/lib/training_route_manager.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ def requires_study_mode?
       TRAINING_ROUTE_ENUMS[:assessment_only],
       TRAINING_ROUTE_ENUMS[:pg_teaching_apprenticeship],
       TRAINING_ROUTE_ENUMS[:opt_in_undergrad],
+      TRAINING_ROUTE_ENUMS[:hpitt_postgrad],
```
Comment (archferns) [inline app/lib/training_route_manager.rb:43]: Note to reviewer: Assuming we don't the route to be exactly like AO only at the moment

File: app/models/provider.rb [hunk 2]
```diff
@@ -18,4 +18,9 @@ class Provider < ApplicationRecord
   def code=(cde)
     self[:code] = cde.to_s.upcase
   end
+
+  def hpitt_postgrad?
+    # An arbitrary provider set here until we receive a list of teach first providers
```
Comment (defong) [inline app/models/provider.rb:23]:
```suggestion
    # TODO: An arbitrary provider set here until we receive a list of teach first providers
```

File: app/views/trainees/training_routes/_form_fields.html.erb [hunk 3]
```diff
@@ -1,16 +1,20 @@
 <%= f.govuk_radio_buttons_fieldset(:training_route, legend: { size: "l", text: t("components.page_titles.trainees.training_routes.edit"), tag: "h1" }) do %>
 
-  <%= f.govuk_radio_button :training_route, TRAINING_ROUTE_ENUMS[:assessment_only], label: { text: t("activerecord.attributes.trainee.training_routes.assessment_only") }, link_errors: true %>
+  <% if current_user.provider.hpitt_postgrad? %>
```
Comment (defong) [inline app/views/trainees/training_routes/_form_fields.html.erb:3]:
```suggestion
  <% if current_user.hpitt_provider? %>
```

File: config/initializers/training_routes.rb [hunk 4]
```diff
@@ -47,11 +47,11 @@
 }.freeze
 
 TRAINING_ROUTES_FOR_TRAINEE = TRAINING_ROUTES.select { |training_route|
-  TRAINING_ROUTE_ENUMS.values_at(:assessment_only, :provider_led_postgrad, :early_years_undergrad, :school_direct_tuition_fee, :school_direct_salaried, :early_years_assessment_only, :early_years_salaried, :early_years_postgrad, :pg_teaching_apprenticeship).include? training_route
+  TRAINING_ROUTE_ENUMS.values_at(:assessment_only, :provider_led_postgrad, :early_years_undergrad, :school_direct_tuition_fee, :school_direct_salaried, :early_years_assessment_only, :early_years_salaried, :early_years_postgrad, :pg_teaching_apprenticeship, :hpitt_postgrad).include? training_route
 }.freeze
 
 TRAINING_ROUTES_FOR_COURSE = TRAINING_ROUTES.select { |training_route|
-  TRAINING_ROUTE_ENUMS.values_at(:provider_led_postgrad, :school_direct_tuition_fee, :school_direct_salaried, :pg_teaching_apprenticeship).include? training_route
+  TRAINING_ROUTE_ENUMS.values_at(:provider_led_postgrad, :school_direct_tuition_fee, :school_direct_salaried, :pg_teaching_apprenticeship, :hpitt_postgrad).include? training_route
```
Comment (defong) [inline config/initializers/training_routes.rb:54]:
```suggestion
  TRAINING_ROUTE_ENUMS.values_at(:provider_led_postgrad, :school_direct_tuition_fee, :school_direct_salaried, :pg_teaching_apprenticeship).include? training_route
```

Don't think `:hpitt_postgrad` is applicable

File: app/views/trainees/training_routes/_form_fields.html.erb (hunk 3 above)
Comment (archferns) [reply app/views/trainees/training_routes/_form_fields.html.erb:3]: Would need to delegate this to the user in that case

File: config/initializers/training_routes.rb (hunk 4 above)
Comment (archferns) [reply config/initializers/training_routes.rb:54]: Not having this as a route will break on https://github.com/DFE-Digital/register-trainee-teachers/blob/7d7da41540282ec51eed952c6aee5e82c908c468/app/lib/hpitt.rb#L51
We might need that one to be amended as well, but unsure of what's the right approach.

Comment (defong) [reply config/initializers/training_routes.rb:54]: That's pointless assumption...

https://github.com/DFE-Digital/register-trainee-teachers/blob/7d7da41540282ec51eed952c6aee5e82c908c468/app/services/teacher_training_api/import_course.rb#L68-L78

Comment (archferns) [reply config/initializers/training_routes.rb]: Updated.

File: spec/factories/trainees.rb [hunk 5]
```diff
@@ -293,7 +293,7 @@
     end
 
     trait :with_related_courses do
-      training_route { (TRAINING_ROUTES_FOR_TRAINEE.keys & TRAINING_ROUTES_FOR_COURSE.keys).sample }
+      training_route { (TRAINING_ROUTES_FOR_TRAINEE.keys & TRAINING_ROUTES_FOR_COURSE.keys - ["hpitt_postgrad"]).sample }
```
Comment (defong) [inline spec/factories/trainees.rb:296]:
```suggestion
      training_route { TRAINING_ROUTES_FOR_COURSE.keys.sample }
```

File: app/lib/hpitt.rb [hunk 6]
```diff
@@ -48,7 +48,6 @@ def find_course(trainee, csv_row)
         duration_in_years: csv_row["Duration"].scan(/\d+/).first.to_i,
         start_date: Date.parse(csv_row["Course start date"]),
         name: csv_row["ITT Subject 1"],
-        route: trainee.training_route,
       )
 
       case potential_courses.count
```
Comment (defong) [inline app/lib/hpitt.rb:53]: `potential_courses` looks buggy

**Review by defong — APPROVED** [review]

**Review by darokel — APPROVED** [review]
Works nicely on review app 👍

**Review by defong — APPROVED** [review]

**Comment by bbelward** [comment]
Working well for me too, nice work :)

**Pull request merged**